src/OpenSearch/Serializers/SmartSerializer.php

```diff
@@ -56,8 +56,17 @@ public function serialize($data): string
      */
     public function deserialize(?string $data, array $headers)
     {
-        if (isset($headers['content_type']) === true) {
-            if (strpos($headers['content_type'], 'json') !== false) {
+        if (isset($headers['Content-Type'])) {
+            $contentType = $headers['Content-Type'];
+        } elseif (isset($headers['content_type'])) {
+            $contentType = $headers['content_type'];
```

Comment:
Interesting. We should look at why we were checking `content_type`  and not `content-type` already. 🤔

Reply:
Also, array keys are case sensitive so we should see what Guzzle does to normalize them.

Reply:
we could change them to small case using https://www.php.net/manual/en/function.array-change-key-case.php to small 🤔

Reply:
Ah looks like `SmartSerializer` had been depending on `$response['transfer_stats']` rather than response headers. I think we should add a comment about `'content_type` being there for backwards compatibility.

Reply:
> we could change them to small case using https://www.php.net/manual/en/function.array-change-key-case.php to small 🤔

We should probably do this.